AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/SQLStatement+Condition.swift

```diff
@@ -65,6 +64,21 @@ private func translateQueryPredicate(from modelSchema: ModelSchema,
         }
     }
 
+    func resolveColumn(_ operation: QueryPredicateOperation) -> String {
```

Comment:
This function is ambiguous, because if you have a namespace, it has the potential to get overwritten if you also have a modelField. (e.g. if line 74 get executed, then line 70 will also get executed -- this seems inefficient)

Is it possible to write some unit tests to demonstrate the intent of this function?

Reply:
e.g.
- `testTranslateQueryPredicateWithNameSpaceWhenFieldNameSpecified` 

hits 
```swift
if let namespace = namespace {
    column = String(namespace).quoted() + "." + column
}
```
but skips
```swift
if let modelField = modelSchema.field(withName: operation.field) { // modelField doesn't return
    if let namespace = namespace {
        column = modelField.columnName(forNamespace: String(namespace)) // execute this line because name space given
    } else {
        column = modelField.columnName()
    }
}
```
because explicit filename is given like this
```swift
let predicate = field("commentPostId") == "postID"
```

- `testTranslateSingleQueryPredicateWithNamespace`

hits both
```swift
if let namespace = namespace {
    column = String(namespace).quoted() + "." + column
}
```
and
```swift
if let modelField = modelSchema.field(withName: operation.field) { // modelField is return
    if let namespace = namespace {
        column = modelField.columnName(forNamespace: String(namespace)) // execute this line because name space given
    } else {
        column = modelField.columnName()
    }
}
```
because it's using CodingKey
```swift
let post = Post.keys
let predicate = post.id != nil
```

- `testTranslateSingleQueryPredicate`

skips
```swift
if let namespace = namespace {
    column = String(namespace).quoted() + "." + column
}
```
but hits
```swift
if let modelField = modelSchema.field(withName: operation.field) { // modelField is return
    if let namespace = namespace {
        column = modelField.columnName(forNamespace: String(namespace))
    } else {
        column = modelField.columnName() // execute this line because namespace not given
    }
}
```

Reply:
I think you may have misunderstood me.  Lets start over.

Is there a cleaner way to write the code so that when we enter the function, we set `column` once and only once?  We shouldn't need to make column a `var`.  If needed, we could break this out into separate functions.

Reply:
I came up with code like this, return a string if it falls into each of the four cases, the force-unwrapping is safe because I make sure that it's not nil
```swift
func resolveColumn(_ operation: QueryPredicateOperation) -> String {
    let namespace = namespace
    let modelField = modelSchema.field(withName: operation.field)

    if namespace != nil && modelField != nil {
        return modelField!.columnName(forNamespace: String(namespace!))
    } else if namespace == nil && modelField != nil {
        return modelField!.columnName()
    } else if namespace != nil && modelField == nil {
        return String(namespace!).quoted() + "." + operation.field.quoted()
    } else {
        return operation.field.quoted()
    }
}
```

Reply:
The code block you provided seems like it would work, and i agree the force-unwrapping is safe, but... it can be written in the following manner:
```
        let modelField = modelSchema.field(withName: operation.field)
        if let namespace = namespace, let modelField = modelField {
            return modelField.columnName(forNamespace: String(namespace))
        } else if let modelField = modelField {
            return modelField.columnName()
        } else if let namespace = namespace {
            return String(namespace).quoted() + "." + operation.field.quoted()
        }
        return operation.field.quoted()
```